src/main/java/com/royalrangers/service/UserService.java

```diff
@@ -253,31 +253,53 @@ private TempUser createTempUser() {
         tempUser.setSection(user.getSection());
 
         tempUserRepository.save(tempUser);
-        return tempUser;
+    }
+
+    public boolean checkTempUser(UserUpdateDto tempUser) {
+        User user = getAuthenticatedUser();
+
+        return tempUser.getBirthDate().equals(user.getBirthDate()) && tempUser.getCityId().equals(user.getCity().getId()) && tempUser.getCountryId().equals(user.getCountry().getId()) &&
+                tempUser.getPlatoonId().equals(user.getPlatoon().getId()) && tempUser.getRegionId().equals(user.getRegion().getId()) && tempUser.getSectionId().equals(user.getSection().getId()) &&
+                tempUser.getFirstName().equals(user.getFirstName()) && tempUser.getLastName().equals(user.getLastName()) && tempUser.getGender().equals(user.getGender()) &&
+                tempUser.getTelephoneNumber().equals(user.getTelephoneNumber()) && tempUser.getUserAgeGroup().equals(user.getUserAgeGroup()) &&
+                tempUser.getUserRank().equals(user.getUserRank());
```

Comment:
Name of this method is not very clear. Possible variant - isTempUserEqualsUser()
And add comment to the method:
 /**
    Compares temp_user to user - if they are the same we should delete temp_user
  */